src/AssetRegistrar.php

```diff
@@ -0,0 +1,514 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Assets;
+
+use Yiisoft\Aliases\Aliases;
+use Yiisoft\Assets\Exception\InvalidConfigException;
+
+use function array_key_exists;
+use function array_values;
+use function get_class;
+use function gettype;
+use function is_array;
+use function is_int;
+use function is_object;
+use function is_string;
+use function sprintf;
+
+/**
+ * AssetRegistrar registers asset files from a bundle considering dependencies.
+ *
+ * @psalm-type CssFile = array{0:string,1?:int}&array
+ * @psalm-type CssString = array{0:mixed,1?:int}&array
+ * @psalm-type JsFile = array{0:string,1?:int}&array
+ * @psalm-type JsString = array{0:mixed,1?:int}&array
+ */
+final class AssetRegistrar
```

Comment:
Mark class as `@internal`. Class should be used only into package.

Reply:
Done.